content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+`toBe` is just one example of a function called a [matcher](https://jestjs.io/docs/using-matchers).
+A matcher is a function we use to compare values in Jest.
+
+So we can combine this with the test we wrote earlier, to get our first complete test case:
+
+```js
+test("converts 1 to an ordinal number", function () {
+  expect(getOrdinalNumber(1)).toBe("1st");
+});
+```
+
+### 👟 Running tests
+
+We can try running the file index.test.js with node in the following way:
+
+```bash
+node index.test.js
+```
+
+but we get an error:
+
+```bash
+ReferenceError: test is not defined
+```
+
+This is because `test` isn’t defined anywhere in the file.
+
+So we need to execute this file so that the Jest API is available in our file. We can do this by adding executing the test file using Jest: we do this using an npm script.
+
+The "scripts" section of the `package.json` is where we can write useful commands we'll use in our project. We can add a "scripts" section to the `package.json` so that it reads as follows:
+
+```json {linenos=table,hl_lines=["4-6"],linenostart=1}
+{
+  "name": "week-4-test-example",
+  "description": "An example application showing how to write tests using the jest framework",
+  "scripts": {
+    "test": "jest"
+  },
+  "devDependencies": {
+    "jest": "^29.5.0"
+  }
+}
+```
+
+Finally, we'll need to run our tests.
+Now we can run the command `npm test`.
+
+When we execute the command, `npm test`, the command will look inside the "scripts" section of the package.json and look up the commmand for "test" - in this case, "jest".
+
+### Understanding test feedback
+
+We currently have a project structure like this:
+
+```raw
+week-4-test-example
+├── get-ordinal-number.test.js
+├── package.json
+├── package-lock.json
+└── node_modules
+
+1 directory, 3 files
+```
+
+And `get-ordinal-number.test.js`
+looks like this
+
+```js
+test("converts 1 to an ordinal number", function () {
+  expect(getOrdinalNumber(1)).toBe("1st");
+});
+```
+
+After running the test above, we should get feedback indicating whether or not the test has passed.
+
+{{<note type="exercise" title="exercise">}}
+Predict what the test feedback will say when the test above is executed.
+{{</note>}}
+
+### 🚢 Defining the function
+
+At the moment, our test feedback gives the following:
+
+![test-reference-error](test-reference-error.png)
+
+The test code is throwing a **ReferenceError**.
+
+We can define `getOrdinalNumber` in our test file.
+
+```js
+function getOrdinalNumber() {}
+
+test("converts 1 to an ordinal number", function () {
+  expect(getOrdinalNumber(1)).toBe("1st");
+});
+```
+
+Now we can run the tests again and check the test feedback.
+
+### Assertion errors
+
+We now get the following feedback:
+
+![test-feedback-fail](test-feedback-fail.png)
+
+Jest tells us 3 main things:
+
+1. The test case that failed
+2. The targetOutput and the currentOutput
```

Comment:
```suggestion
2. The target output and the current output
```

I'm not sure camel-casing these when we're not referring to actual variables makes sense?